### DISCUSSION TITLE
Target conflict

### DISCUSSION THREAD

**Discussion by dmuysers** [modula3/cm3#1169] [Q&A]
After installing Windows 11 on my new desktop 64-bit computer, I downloaded and installed, as instructed,
cm3-dist-AMD64_NT-d5.11.4.7z
Then testing a simple program on the  vcvarsall.bat 64 command prompt:
MODULE Main;
IMPORT RTIO;
BEGIN
  IF BITSIZE (WIDECHAR) = 32 THEN RTIO.PutText ("TRUE") ELSE RTIO.PutText ("FALSE") END;  
END Main.
I got
Main.mo : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64'
What is the so-called "module machine type" ?

**Comment by dmuysers** [comment]
Trying to find a solution, I removed cm3 and my old VS2019 community edition,
installed VS2022 community edition, and tried to reinstall cm3:
D:\Users\Dirk\scratch\build>python3 ..\cm3\scripts\concierge.py install --prefix C:\cm3
..\cm3\scripts\concierge.py install --prefix C:\cm3
cmake -S D:\Users\Dirk\scratch\cm3\bootstrap -B D:\Users\Dirk\scratch\build -DCMAKE_INSTALL_PREFIX=C:\cm3 -A x64
-- Building for: NMake Makefiles
CMake Error at CMakeLists.txt:2 (project):
  Generator
    NMake Makefiles
  does not support platform specification, but platform
    x64
  was specified.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage

**Reply by VictorMiasnikov** [reply]
> } machine type 'x86' 

It is very strange...

> } on the vcvarsall.bat 64 command prompt

Are You sure?  ( I'm personal time to time run 32bit environment by mistake)


---

Ok:  I am on notebook. Level energy on battery is near to low

I'm simply public good cm3 And Co with {{ 50/50 worked LLVM13 }}

**Reply by VictorMiasnikov** [reply]
P.S.

cm3-dist-AMD64_NT-d5.11.4.7z   -- is very and very good and tested distributive.  It's tested exactly _with_ VS 2022 on GitHub Action environment

**Comment by VictorMiasnikov** [comment]
Please see, 

https://github.com/VictorMiasnikov/cm3/releases/tag/d5.11.9-ZZYYXX-20230928_16-50

[Password_is_c9ac5ff.__cm3-min-AMD64_NT-d5.11.9-VC143.2022.-2023-09-28_16-50__FIXed__by__VVM__2023-09-29_15-34__16bit__Unicode__AMD64_NT__Target__cm3-c9ac5ff.for.xDelta.7z_In.7z](https://github.com/VictorMiasnikov/cm3/releases/download/d5.11.9-ZZYYXX-20230928_16-50/Password_is_c9ac5ff.__cm3-min-AMD64_NT-d5.11.9-VC143.2022.-2023-09-28_16-50__FIXed__by__VVM__2023-09-29_15-34__16bit__Unicode__AMD64_NT__Target__cm3-c9ac5ff.for.xDelta.7z_In.7z)


2025-04-10 :

 You can try this:

`cm3-all-AMD64_NT-d5.11.9-VC143.2022.-2024-11-18_21-50__FIXed__by__VVM__2025-02-27_21-10__16bit__Unicode__AMD64_NT__Target__cm3-08d84d1.7z
`
 Or this:

`cm3-all-AMD64_NT-d5.11.9-VC143.2022.-2024-11-18_21-50__FIXed__by__VVM__2025-03-28_18-47__16bit__Unicode__AMD64_NT__Target__cm3-08d84d1.msi
`
 from

https://github.com/VictorMiasnikov/cm3/releases/tag/d5.11.9-ZZYYXX-20241118_21-50
